projects/element-ng/chat-messages/si-attachment-list.component.ts

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) Siemens 2016 - 2025
+ * SPDX-License-Identifier: MIT
+ */
+import { booleanAttribute, Component, inject, input, output, TemplateRef } from '@angular/core';
+import { SiIconComponent } from '@siemens/element-ng/icon';
+import { SiModalService } from '@siemens/element-ng/modal';
+import { SiTranslatePipe, t } from '@siemens/element-translate-ng/translate';
+
+/**
+ * Attachment item interface for file attachments in chat messages.
+ *
+ * @experimental
+ */
+export interface Attachment {
+  /** Unique identifier for the attachment */
+  id: string;
+  /** File name */
+  name: string;
+  /** Show a preview of the attachment by providing a template that is shown in a modal when clicked (optional) */
+  previewTemplate?: TemplateRef<any> | (() => TemplateRef<any>);
```

Comment:
How do you envision the use here. Can we add an example?

Reply:
Added to the example (was already mentioned in the TSDOC but very briefly). Basically it allows a consumer to show a file preview when clicking on the attachment pill, it makes sense in the chat use-case.